lib/src/access.ts

```diff
@@ -301,3 +203,15 @@ export class OriginAllowList {
     return this.origins.includes(origin(url));
   }
 }
+
+/**
+ * Converts a KAS endpoint URL to a platform URL.
+ * If the KAS endpoint ends with '/kas', it returns the host url
+ * Otherwise, it returns the original KAS endpoint.
+ */
+function getHostFromEndpoint(endpoint: string): string {
+  // TODO RPC: find a better way to get the right url, otherwise just use the `origin` function
```

Comment:
When `Rewrap` is called the URL received from the TDF manifest has the KAS endpoint sufix: `http://platfrom.com/kas`. The issue is that RPC requires the URL of the platform and not the KAS to create the transport layer.

If the URL is `http://platform.com/kas` we can either remove `/kas` or take the origin. Both would return `http://platform.com`

But if the url is `http://platform.com/v2/kas` than the result will be different and technically we can have cases where either might be correct

Not really sure how to resolve this in a solid way.

Reply:
You should just have to strip off `/kas` and the hostname will be the KAS you need to call.

Reply:
Does that support KAS-only mode? (I only know about it from this mention)

https://docs.google.com/presentation/d/1XYVcviUbrHrsvlOGLZcJQQHN2IJTKyViE2_oR1mDpAU/edit?slide=id.g2d86a436258_0_6#slide=id.g2d86a436258_0_6

Reply:
Updated to strip of trailing `/` and `/kas`